spec/policies/task_policy_spec.rb

```diff
@@ -32,40 +32,12 @@
 
     it { is_expected.to permit_only_actions(generic_user_permissions) }
 
-    context 'when user is admin' do
-      let(:user) { create(:admin) }
-
-      context 'without gpt access token' do
-        it { is_expected.to forbid_only_actions %i[generate_test] }
-      end
-
-      context 'with gpt access token' do
-        before do
-          Settings.open_ai.access_token = 'access_token'
-        end
-
-        after do
-          Settings.open_ai.access_token = nil
-        end
-
-        it { is_expected.to permit_all_actions }
-      end
-    end
-
     context 'when task is from user' do
       let(:task_user) { user }
 
-      context 'without gpt access token' do
-        it { is_expected.to forbid_only_actions %i[generate_test] }
-      end
-
       context 'with gpt access token' do
         before do
-          Settings.open_ai.access_token = 'access_token'
-        end
-
-        after do
-          Settings.open_ai.access_token = nil
+          allow(user).to receive(:personal_openai_api_key).and_return('access_token')
```

Comment:
You could add a variable attribute to the user in line 30 and only set it when needed.

Reply:
Not sure what I need to do here. could you give an example, please?

Reply:
You can compare it to the `task` created in line 8-11.
You have to add an attribute to the `create(:user)` in line 30 (`create(:user, open_ai_key:)`) and then initialize it with nil (`let(:open_ai_key) {nil}`).
In your context you can then just overwrite the let with the desired value